lib/rspec/mocks/proxy.rb

```diff
@@ -371,20 +371,34 @@ def original_method_handle_for(message)
 
         return super unless unbound_method
         unbound_method.bind(object)
+      rescue TypeError
+        if RUBY_VERSION == '1.8.7'
```

Comment:
That's a good point.  It's the general pattern we use but when there's a reason not to follow it (as you've explained well) then it's fine not to -- so let's stick with what you have.

Can you add `# :nocov:` before and after this section so that it isn't considered for simplecov code coverage?